### DISCUSSION TITLE
Action on hardware button click

### DISCUSSION THREAD

**Discussion by Gitavit** [Blackymas/NSPanel_HA_Blueprint#2449] [Q&A]
Hi, 
I just tested this blueprint and really love it! thank you for your great contributions!
There is just one thing which I hardly miss: UI elements to controll the tilt angle of blinds. It looks like this is not (yet) availeable. Therefore I tried to use the physical buttons in the following way:

Left Button:
- Hold: cover.open_cover
- Click: cover.open_cover_tilt_incremental (20%)

Right Button: 
- Hold: cover.close_cover
- Click: cover.close_cover_tilt_incremental (20%)

While it's very easy to setup the hold-action using the blueprint, I couldn't figure out how to do the same for the click-action. THe blueprint only allows to link an entity to the button click, but not an action.

Do you know, if that's possible with some sort of manual workaround?

BR David

**Comment by edwardtfn** [comment]
As a workaround you can create an automation on your Home Assistant using the sensors related to the button and the current page.
I can take a look at some implementation for that in the short term, but then we are talking about some weeks still, as I'm travelling right now and the version on dev still quite buggy.

**Reply by Gitavit** [reply]
Thank you for your reply! I'll try it solve it with an automation. However, I think this will be a bit tricky, as the sensor does not differentiete between short an long press. Just between on and off. Let's see, if I can make it work witth some thresholds.

**Reply by edwardtfn** [reply]
I'm pretty sure the hardware buttons will always trigger events for both "short_click" and "long_click". You can go to Developer Tools > Events tab and try to listen to the events from your panel and do some test.
